src/test/emulators/functionsEmulatorRuntime.spec.ts

```diff
@@ -42,10 +42,52 @@ function InvokeRuntimeWithFunctions(
   });
 }
 
-function _is_verbose(runtime: FunctionsRuntimeInstance): void {
-  runtime.events.on("log", (el: EmulatorLog) => {
-    process.stdout.write(el.toPrettyString() + "\n");
+/**
+ * Three step process:
+ *   1) Wait for the runtime to be ready.
+ *   2) Call the runtime with the specified bundle and collect all data.
+ *   3) Wait for the runtime to exit
+ */
+async function CallHTTPSFunction(
+  runtime: FunctionsRuntimeInstance,
+  frb: FunctionsRuntimeBundle,
+  options: any = {},
+  requestData?: string
+): Promise<string> {
+  await runtime.ready;
+  const dataPromise = new Promise<string>((resolve, reject) => {
```

Comment:
Shall we just use the `request` module instead? It's already in `dependencies` and it provides a nicer callback of `function (err, response, body)` so you don't have to do the manual string concat stuff here.

P.S. The current implementation fails to propagate errors reading the body (a.k.a. `res.on("error", reject)` on stream).

Reply:
Good idea!